src/core/src/CubicSpline.cpp

```diff
@@ -0,0 +1,175 @@
+    
+    if(m_velocities.size() > 2){
+        if(!this->computeIntermediateVelocities())
+            return false;
+    }
+    
+    for(size_t i = 0; i < m_coefficients.size(); ++i){
+        m_coefficients[i](0) = m_y(i);
+        m_coefficients[i](1) = m_velocities(i);
+        m_coefficients[i](2) = ( 3*(m_y(i+1) - m_y(i))/m_T(i) - 2*m_velocities(i) - m_velocities(i+1) )/m_T(i);
+        m_coefficients[i](3) = ( 2*(m_y(i) - m_y(i+1))/m_T(i) + m_velocities(i) + m_velocities(i+1) )/std::pow(m_T(i), 2);
+    }
+    
+    return true;
+}
+
+bool iDynTree::CubicSpline::computeIntermediateVelocities()
+{
+    Eigen::MatrixXd A(m_velocities.size()-2, m_velocities.size()-2);
+    Eigen::VectorXd b(m_velocities.size()-2);
+    
+    A.setZero();
+    
+    A(0,0) = 2*(m_T(0) + m_T(1));
+    b(0) = ( std::pow(m_T(0), 2)*(m_y(2) - m_y(1)) + std::pow(m_T(1), 2)*(m_y(1) - m_y(0)) )*3/(m_T(0)*m_T(1)) - m_T(1)*m_velocities(0);
+    
+    if(m_velocities.size() > 3){
+        A(0,1) = m_T(0);
+        for(int i = 1; i < A.rows()-1; ++i){
+            A(i,i-1) = m_T(i+1);
+            A(i,i) = 2*(m_T(i) + m_T(i+1));
+            A(i,i+1) = m_T(i);
+            
+            b(i) = ( std::pow(m_T(i), 2)*(m_y(i+2) - m_y(i+1)) + std::pow(m_T(i+1), 2)*(m_y(i+1) - m_y(i)) )*3/(m_T(i)*m_T(i+1));
+        }
+        size_t T = m_T.size() - 1; 
+        A.bottomRightCorner<1,2>() << m_T(T), 2*(m_T(T) + m_T(T-1));
+        b.tail<1>() << ( std::pow(m_T(T-1), 2)*(m_y(T+1) - m_y(T)) + std::pow(m_T(T), 2)*(m_y(T) - m_y(T-1)) )*3/(m_T(T-1)*m_T(T)) - m_T(T-1)*m_velocities(T+1);
+    }
+    
+   iDynTree::toEigen(m_velocities).segment(1, m_velocities.size()-2) = A.colPivHouseholderQr().solve(b);
+    
+    return true;
+}
+
+bool iDynTree::CubicSpline::computePhasesDuration()
+{
+    for (size_t i = 0; i < m_time.size() - 1; ++i){
+        
+        m_T(i) = m_time(i+1) - m_time(i);
+        
+        if(m_T(i) == 0){
+            std::cerr << "[CUBICSPLINE] Two consecutive points have the same time coordinate." << std::endl; //For stability purposes, the matrix below may not be invertible
+            return false;
+        }
+        
+        if(m_T(i) < 0){
+            std::cerr << "[CUBICSPLINE] The input points are expected to be consecutive, strictly increasing in the time variable." << std::endl; //For stability purposes
+            return false;
+        }
+        
+    }
+    return true;
+}
+
+void iDynTree::CubicSpline::setInitialConditions(double initialVelocity, double initialAcceleration)
+{
+    m_v0 = initialVelocity;
+    m_a0 = initialAcceleration;
+}
+
+void iDynTree::CubicSpline::setFinalConditions(double finalVelocity, double finalAcceleration)
+{
+    m_vf = finalVelocity;
+    m_af = finalAcceleration;
+}
+
+double iDynTree::CubicSpline::evaluatePoint(double t)
+{
+    if(m_time.size() == 0){
+        std::cerr << "[CUBICSPLINE] First you have to load data! The returned data should not be considered." << std::endl; 
+        return 1e19;
```

Comment:
If you want to return the biggest double, you can use `std::numeric_limits<double>::max()` (you need to include `<limits>`). However, I don't know if returning the maximum possible double is the wisest choice here.